### PR TITLE
sharebymail: remove ILogger and upgrade to PHP 8 syntax

### PR DESCRIPTION
**Pull request by Altahrim**
Ref: #32127 

## Summary
 - Remove ILogger from `sharebymail` application
 - Upgrade to PHP 8 syntax (types, promoted properties)

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/sharebymail/tests/ShareByMailProviderTest.php [hunk 1]
```diff
@@ -247,9 +244,7 @@ public function testCreateSendPasswordByMailWithoutEnforcedPasswordProtection()
 
 		$this->mailer->expects($this->never())->method('send');
 
-		$this->assertSame('shareObject',
-			$instance->create($share)
-		);
+		$this->assertInstanceOf(IShare::class, $instance->create($share));
```
Comment (come-nc) [inline apps/sharebymail/tests/ShareByMailProviderTest.php:247]: This feels a bit weak as an assertion, create is typed to return an IShare, no?

**Review by come-nc — COMMENTED** [review]
I would feel better if tests are hardened, otherwise good.

File: apps/sharebymail/tests/ShareByMailProviderTest.php (hunk 1 above)
Comment (Altahrim) [reply apps/sharebymail/tests/ShareByMailProviderTest.php:247]: Yes, it basically test if above `expects` behave as expected and it is able to build the object.
The refactor don't really change the test.

Do you have another idea about something to test?

**Review by ArtificialOwl — APPROVED** [review]

File: apps/sharebymail/tests/ShareByMailProviderTest.php (hunk 1 above)
Comment (come-nc) [reply apps/sharebymail/tests/ShareByMailProviderTest.php:247]: That the object returned is the one returned by createShareObject, or that its id is 42?

Comment (Altahrim) [reply apps/sharebymail/tests/ShareByMailProviderTest.php]: Done!

**Review by come-nc — APPROVED** [review]
Apply the same change to the other modified tests please, then all good.

**Pull request merged**